primitives/avn-common/src/event_discovery.rs

```diff
@@ -14,6 +15,19 @@ pub struct EthBlockRange {
     pub length: u32,
 }
 
+impl EthBlockRange {
+    pub fn next_range(&self) -> EthBlockRange {
+        EthBlockRange {
+            start_block: self.start_block.saturating_add(self.length),
+            length: self.length,
+        }
+    }
+    pub fn range(&self) -> (u32, u32) {
+        let end_block = self.start_block.saturating_add(self.length).saturating_less_one();
+        (self.start_block, end_block)
+    }
+}
+
 #[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, Default, TypeInfo, MaxEncodedLen)]
 pub struct DiscoveredEvent {
```

Comment:
```suggestion
pub struct EthereumEventsReport {
```

Reply:
I will say no to the suggestion. Its a single event that has been discovered and the block it was included.